src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -55,11 +61,49 @@ namespace
 
         return StatusCode::Success;
     }
+
+    int get_or_create_hostpolicy_context(
+        hostpolicy_init_t &hostpolicy_init,
+        const arguments_t &args,
+        bool breadcrumbs_enabled,
+        std::shared_ptr<hostpolicy_context_t> &context)
+    {
+        context = g_context_weak.lock();
+        if (context != nullptr)
+        {
+            // [TODO] Validate the current context is acceptable for this request
+            trace::info(_X("Host context has already been initialized"));
+            return StatusCode::Success;
+        }
+
+        {
+            std::lock_guard<std::mutex> lock{ g_context_lock };
+            context = g_context_weak.lock();
+            if (context != nullptr)
+            {
+                trace::info(_X("Host context has already been initialized"));
+                return StatusCode::Success;
+            }
+
+            std::unique_ptr<hostpolicy_context_t> context_local(new hostpolicy_context_t());
+            int rc = context_local->initialize(hostpolicy_init, args, breadcrumbs_enabled);
+            if (rc != StatusCode::Success)
+                return rc;
+
+            assert(g_context == nullptr);
+            g_context = std::move(context_local);
+            g_context_weak = g_context;
```

Comment:
It is not clear to me what value this duplication is buying.  Why do we need to global pointers to the same object?

Reply:
I was thinking something about shutdown when I did this that didn't really end up making sense. Removed.